### PR TITLE
Remove identical but different named functions

### PR DESCRIPTION
**Pull request by 9rnsr**

### CODE REVIEW

File: src/dsymbol.d [hunk 1]
```diff
@@ -835,6 +819,13 @@ public:
         return p ? p.isAggregateDeclaration() : null;
     }
 
+    // is this a member of a ClassDeclaration?
```
Comment (WalterBright) [inline src/dsymbol.d]: What's the purpose of moving this function?

File: src/dsymbol.d [hunk 2]
```diff
@@ -819,6 +810,15 @@ public:
         return p ? p.isAggregateDeclaration() : null;
     }
 
+    // is this a member of an AggregateDeclaration?
```
Comment (WalterBright) [inline src/dsymbol.d]: Why move the location of the function?

File: src/dsymbol.d (hunk 1 above)
Comment (9rnsr) [inline src/dsymbol.d]: `Dsymbol.isClassMember` is now a specialized version of `Dsymbol.isMember`. Moving the function is good to represent those relations.

File: src/dsymbol.d (hunk 2 above)
Comment (9rnsr) [inline src/dsymbol.d]: `Dsymbol.isMember2` is now a tweaked version of `Dsymbol.isMember`. Moving the function is good to represent those relations.

**Comment by Geod24** [comment]
Any chance you could seize the occasion to add proper documentation to these functions ?

**Comment by 9rnsr** [comment]
@Geod24 I added a detailed ddoc comment to explain what is returned from `parent`, `toParent()`, and `toParent2()`. The ddocs on `isMember` and `isMember2` refer it.

**Comment by WalterBright** [comment]
Auto-merge toggled on

**Pull request merged**